googletest/src/gtest-internal-inl.h

```diff
@@ -74,6 +74,8 @@ GTEST_DECLARE_bool_(death_test_use_fork);
 namespace testing {
 namespace internal {
 
+enum class GTestColorMode { kNo, kYes, kAnsi };
```

Comment:
I'm worried about the difference between `yes` and `ansi`. Both of these are used to turn on colors, but in different ways. If we were to design GoogleTest from scratch, we'd probably want to provide a flag with possible values such as `ansi`, `no`, `auto`, and `not_ansi` (probably something more informative for the last one).

Instead of expanding the existing flag, could we add another one for using ANSI color escape sequences? I'm okay with it being a force flag (on/off) or a feature flag (on/off/auto).

Reply:
The enum is made with the logic 0 - coloring is disabled, !0 - coloring is enabled, 1 or 2 - it's a detail how it is enabled.

Reply:
I could make it as another flag `--gtest_color_ansi` and keep `--gtest_color` unchanged. but did not want to add yet another environment variable and to make a larger changeset.

Reply:
I'd recommend adding another flag, e.g. something like `--gtest_use_ansi_color` and keep the existing one unchanged.
Most users shouldn't care about the new flag, but by providing it, we leave an escape hatch for those that fail to auto-detect ANSI colors.